pkg/tsdb/mysql/macros.go

```diff
@@ -1,31 +1,43 @@
 package mysql
 
 import (
+	"errors"
 	"fmt"
 	"regexp"
 	"strings"
 
 	"github.com/grafana/grafana/pkg/components/gtime"
+	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/tsdb"
 	"github.com/grafana/grafana/pkg/tsdb/sqleng"
 )
 
 const rsIdentifier = `([_a-zA-Z0-9]+)`
 const sExpr = `\$` + rsIdentifier + `\(([^\)]*)\)`
 
+var restrictedRegExp = regexp.MustCompile(`(?im)([\s]*show[\s]+grants|[\s,]session_user\([^\)]*\)|[\s,]current_user(\([^\)]*\))?|[\s,]system_user\([^\)]*\)|[\s,]user\([^\)]*\))([\s,;]|$)`)
```

Comment:
The following feels like it wont work and/or have unexpected outcome. For example existing queries using `select col as current_user` would result in error and be a breaking change.

I think it's hard to fully restrict this using regex since there may be other functions we don't know about.

Reply:
Think this should be fine (very minor breaking change) and a good thing to add (especially restricting show grants). It is the first time we start parsing and restricting query text but think it's a good thing to do.